atomic_reactor/plugins/pre_add_image_content_manifest.py

```diff
@@ -130,13 +135,21 @@ def icm(self):
 
         :return: dict, the ICM as a Python dict
         """
-        if not self.remote_sources and self._icm is None:
+        if not self.remote_sources and not self.pnc_artifact_ids and self._icm is None:
             self._icm = deepcopy(self.minimal_icm)
         elif self._icm is None:
-            request_ids = [remote_source['request_id'] for remote_source in self.remote_sources]
-            self._icm = self.cachito_session.get_image_content_manifest(
-                request_ids
-            )
+            if self.remote_sources:
+                request_ids = [remote_source['request_id'] for remote_source in self.remote_sources]
+                self._icm = self.cachito_session.get_image_content_manifest(
+                    request_ids
+                )
+            if self.pnc_artifact_ids:
+                if self._icm is None:
+                    self._icm = deepcopy(self.minimal_icm)
+                purl_specs = self.pnc_util.get_artifact_purl_specs(self.pnc_artifact_ids)
+                for purl_spec in purl_specs:
+                    self._icm['image_contents'].append({'purl': purl_spec})
```

Comment:
The content manifest from Cachito is sorted by purl. I don't think that's a requirement, it was mostly done just to keep the order deterministic. It does seem that the middleware purls will be added in a deterministic order.

Probably no changes needed, just pointing it out :)

Reply:
That sort of function won't work for this data since it requires sources and dependencies also. And yes, we don't need it here since this is deteministic.